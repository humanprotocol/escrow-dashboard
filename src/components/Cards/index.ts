export { CardTextBlock } from './CardTextBlock';
export { CardBlockWithChildren } from './CardBlockWithChildren';
export { CardLinkBox } from './CardLinkBox';
