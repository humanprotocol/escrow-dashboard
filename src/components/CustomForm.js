import React, {useState} from 'react';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import Stack from "@mui/material/Stack";

const CustomForm = ({ status, message, onValidated }) => {
    const [email, setEmail] = useState('');

    const handleSubmit = (e) => {
      e.preventDefault();
      email &&
      email.indexOf("@") > -1 &&
      onValidated({
          EMAIL: email,
      });
    }

    return (
      <form className="mc__form" onSubmit={(e) => handleSubmit(e)}>
        {status === "sending" && (
          <div className="mc__alert mc__alert--sending" style={{marginBottom: '20px'}}>
            sending...
          </div>
        )}
        {status === "error" && (
          <div 
            className="mc__alert mc__alert--error"
            dangerouslySetInnerHTML={{ __html: message }}
            style={{marginBottom: '20px'}}
          />
        )}
        {status === "success" && (
          <div
            className="mc__alert mc__alert--success"
            dangerouslySetInnerHTML={{ __html: message }}
            style={{marginBottom: '20px'}}
          />
        )}

        <Stack
            direction={{ lg: "row", md: "row", sm: "row", xs: "column" }}
            spacing={2}
            sx={{
              width: "100%"
            }}
          >
          <div className="mc__field-container">
            <TextField
                variant="outlined"
                size="small"
                label="Email"
                sx={{width: "calc(100%)",  flexBasis: "100%"}}
                onChange={(e) => setEmail(e.target.value)}
              >
                {email}
            </TextField>
          </div>
            <Button
              type="submit"
              sx={{backgroundColor: '#320A8D', color: 'white', textTransform: 'capitalize', paddingLeft: '40px', paddingRight: '40px', marginLeft: '20px'}}
            >
                Subscribe
            </Button>
          </Stack>
      </form>
    );
};

export default CustomForm;