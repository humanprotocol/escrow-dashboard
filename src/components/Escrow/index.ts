export { EscrowContainer as Escrow } from './Escrow';
export { EscrowFactoryView } from './EscrowFactoryView';
