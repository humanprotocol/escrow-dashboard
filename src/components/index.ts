export { Escrow, EscrowFactoryView } from './Escrow';
export { App, AppNetworkContext } from './App';
export { NetworkSwitcher } from './NetworkSwitcher';
export { CardBlockWithChildren, CardTextBlock } from './Cards';
export { SkeletonLaunched } from './Skeletons';
export { Navbar } from './Navbar';
export { Header } from './Header';
export { Footer } from './Footer';
export { Layout } from './Layout';
export { Search } from './Search';
