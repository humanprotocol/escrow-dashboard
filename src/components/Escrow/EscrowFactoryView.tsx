import * as React from 'react';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import { CardTextBlock, CardLinkBox } from '../Cards';

interface IEscrow {
  count: number;
  address: string;
  latestEscrow: string;
  eventsUrl: string;
  scanner: string;
  pendingEventCount?: number | string;
  bulkTransferEventCount?: number | string;
  intermediateStorageEventCount?: number | string;
}

function getTotal(
  pendingEventCount: number | string,
  bulkTransferEventCount: number | string,
  intermediateStorageEventCount: number | string
): number | string {
  if (
    typeof pendingEventCount === 'string' &&
    typeof bulkTransferEventCount === 'string' &&
    typeof intermediateStorageEventCount === 'string'
  ) {
    return 'N/A';
  }

  return (
    Number(pendingEventCount) +
    Number(bulkTransferEventCount) +
    Number(intermediateStorageEventCount)
  );
}
export const EscrowFactoryView: React.FC<IEscrow> = ({
  count,
  address,
  latestEscrow,
  eventsUrl,
  scanner,
  pendingEventCount = 'N/A',
  bulkTransferEventCount = 'N/A',
  intermediateStorageEventCount = 'N/A',
}): React.ReactElement => {
  const totalEvents = getTotal(
    pendingEventCount,
    bulkTransferEventCount,
    intermediateStorageEventCount
  );
  return (
    <Card variant="outlined">
      <CardContent>
        <CardTextBlock title="Escrow Factory Address" value={address} />
        <CardTextBlock title="Latest Escrow" value={latestEscrow} />
        <CardTextBlock title="Amount Of Escrows" value={count} />
        <CardTextBlock title="Pending Events" value={pendingEventCount} />
        <CardTextBlock
          title="BulkTransfer Events"
          value={bulkTransferEventCount}
        />
        <CardTextBlock
          title="IntermediateStorage Events"
          value={intermediateStorageEventCount}
        />
        <CardTextBlock title="Total Number Of events" value={totalEvents} />
        <CardLinkBox
          url={eventsUrl}
          text={scanner}
          header="All Launched Escrows"
        />
      </CardContent>
    </Card>
  );
};
