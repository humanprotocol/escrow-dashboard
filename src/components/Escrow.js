import {useEffect, useState} from 'react';

import getWeb3 from '../web3';
import { ethers } from 'ethers';
import EscrowFactoryView from './EscrowFactoryView';
import EscrowFactoryABI from '../contracts/EscrowFactoryABI.json';

export default function EscrowContainer({address, scanner, rpcUrl}) {
    const [count, setCount] = useState(0);
    const [contractData, setContractData] = useState(['', '', '']);
    const eventsUrl = `${scanner}/address/${address}#events`;
    
    useEffect(() => {
        
        async function setupEscrow() {
            try {
                const provider = getWeb3(rpcUrl);
                const EscrowFactory = new ethers.Contract(address, EscrowFactoryABI, provider)
                let escrowCount = await EscrowFactory.counter();
                let lastEscrow = await EscrowFactory.lastEscrow();
                let eip = await EscrowFactory.eip20();
                let escrowCounters = await EscrowFactory.escrowCounters(lastEscrow);
                setCount(ethers.utils.formatUnits(escrowCount, 0));
                setContractData([lastEscrow, eip, ethers.utils.formatUnits(escrowCounters, 0)]);

                EscrowFactory.on('Launched', async (eip, escrow) => {
                    console.log(eip, escrow);
                    escrowCount = await EscrowFactory.counter();
                    escrowCounters = await EscrowFactory.escrowCounters(escrow);
                    setCount(ethers.utils.formatUnits(escrowCount, 0));
                    setContractData([escrow, eip, ethers.utils.formatUnits(escrowCounters, 0)]);
                });
            } catch(err) {
                console.log("error_", err);

                alert("Invalid escrow factory");
            }

        }
        setupEscrow();
    }, [address]);

    return (
        <EscrowFactoryView
          count={ count }
          address={address}
          eventsUrl={eventsUrl}
          scanner={scanner}
          contractData={contractData}
        />
    )
}
