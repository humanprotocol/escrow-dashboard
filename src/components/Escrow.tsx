import { useContext, useEffect, useState } from "react";
import { AbiItem } from 'web3-utils'

import getWeb3 from '../web3';
import EscrowFactoryView from './EscrowFactoryView';
import EscrowFactoryABI from '../contracts/EscrowFactoryABI.json';
import { ESCROWFACTORIES_COUNT, ESCROWFACTORY_COUNT } from '../queries';
import AppContext from '../AppNetworkContext';
import { networkMap } from '../constants';
import { countEscrowFactory } from '../utils';

interface Props {
  escrowFactory: string
}
export default function EscrowContainer({ escrowFactory }: Props) {
  const [latestEscrow, setLatestEscrow] = useState("");
  const { network } = useContext(AppContext);

  const { scanner } = networkMap[network];
  const address = networkMap[network].defaultFactoryAddr || escrowFactory;
  const { rpcUrl } = networkMap[network];

  const eventsUrl = `${scanner}/address/${address}#events`;
  const { data } = useQuery(ESCROWFACTORIES_COUNT);

  const { data: dataFactory } = useQuery(ESCROWFACTORY_COUNT, {
    variables: { id: escrowFactory },
    skip: !escrowFactory,
  });

  useEffect(() => {
    async function setupEscrow() {
      try {
        const web3 = getWeb3(rpcUrl);
        const EscrowFactory = new web3.eth.Contract(EscrowFactoryABI as AbiItem[], address);
        const lastEscrow = await EscrowFactory.methods.lastEscrow().call();
        setLatestEscrow(lastEscrow);
      } catch (err) {
        console.error(err);
        alert('Invalid escrow factory');
      }
    }
    setupEscrow();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [address]);

  return (
    <EscrowFactoryView
      count={countEscrowFactory(
        dataFactory?.escrowFactory
          ? [dataFactory.escrowFactory]
          : data?.escrowFactories
      )}
      address={address}
      latestEscrow={latestEscrow}
      eventsUrl={eventsUrl}
      scanner={scanner}
    />
  );
}
