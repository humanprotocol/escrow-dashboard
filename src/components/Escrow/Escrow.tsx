import * as React from 'react';
import { useQuery } from '@apollo/client';

import { getWeb3 } from 'src/helpers';
import {
  // BULK_COUNT,
  ESCROWFACTORIES_COUNT,
  ESCROWFACTORY_COUNT,
  // IS_EVENTS_COUNT,
  // PEVENTS_COUNT,
} from 'src/queries';
import { networkMap } from 'src/constants';
import { countEscrowFactory } from 'src/utils';
import { AppNetworkContext } from 'src/components/App';
import { EscrowFactoryView } from './EscrowFactoryView';
// import { useMultyQuerries } from './hooks';

const EscrowFactoryABI = require('src/contracts/EscrowFactoryABI.json');

interface IEscrowContainer {
  escrowFactory: string;
}

export const EscrowContainer: React.FC<IEscrowContainer> = ({
  escrowFactory,
}): React.ReactElement => {
  const [latestEscrow, setLatestEscrow] = React.useState('');
  const { network } = React.useContext(AppNetworkContext);

  const { scanner } = networkMap[network];
  const address = networkMap[network].defaultFactoryAddr || escrowFactory;
  const { rpcUrl } = networkMap[network];

  const eventsUrl = `${scanner}/address/${address}#events`;
  const { data } = useQuery(ESCROWFACTORIES_COUNT, {
    variables: { skip: 2 },
  });

  // useMultyQuerries();

  const { data: dataFactory } = useQuery(ESCROWFACTORY_COUNT, {
    variables: { id: escrowFactory },
    skip: !escrowFactory,
  });

  React.useEffect(() => {
    async function setupEscrow() {
      try {
        const web3 = getWeb3(rpcUrl);
        const EscrowFactory = new web3.eth.Contract(EscrowFactoryABI, address);
        const lastEscrow = await EscrowFactory.methods.lastEscrow().call();
        setLatestEscrow(lastEscrow);
      } catch (err) {
        console.error(err);
        alert('Invalid escrow factory');
      }
    }
    setupEscrow();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [address]);

  return (
    <EscrowFactoryView
      count={countEscrowFactory(
        dataFactory?.escrowFactory
          ? [dataFactory.escrowFactory]
          : data?.escrowFactories
      )}
      address={address}
      latestEscrow={latestEscrow}
      eventsUrl={eventsUrl}
      scanner={scanner}
    />
  );
};
