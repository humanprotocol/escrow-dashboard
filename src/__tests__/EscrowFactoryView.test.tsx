import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { EscrowFactoryView } from 'src/components/Escrow';

const mock = {
  count: 10,
  address: '63fea823365f1c81fad234abdf5a1f43',
  latestEscrow: 'eaac4d45c3c41f449cf7c94622afacbc',
  eventsUrl: 'test.url',
  scanner: 'test-scanner',
};

describe('when rendered with a `count` prop', () => {
  // const container = render(
  //   <EscrowFactoryView
  //     count={mock.count}
  //     address={mock.address}
  //     latestEscrow={mock.latestEscrow}
  //     eventsUrl={mock.eventsUrl}
  //     scanner={mock.scanner}
  //   />
  // );
  it('should paste it into the component', () => {
    render(
      <EscrowFactoryView
        count={mock.count}
        address={mock.address}
        latestEscrow={mock.latestEscrow}
        eventsUrl={mock.eventsUrl}
        scanner={mock.scanner}
      />
    );
    expect(screen.getByText(mock.count)).toBeInTheDocument();
  });
});

describe('when rendered with a `address` prop', () => {
  it('should paste it into the component', () => {
    render(
      <EscrowFactoryView
        count={mock.count}
        address={mock.address}
        latestEscrow={mock.latestEscrow}
        eventsUrl={mock.eventsUrl}
        scanner={mock.scanner}
      />
    );
    expect(screen.findByLabelText(mock.address)).toBeTruthy();
  });
});

describe('when rendered with a `latestEscrow` prop', () => {
  it('should paste it into the component', () => {
    render(
      <EscrowFactoryView
        count={mock.count}
        address={mock.address}
        latestEscrow={mock.latestEscrow}
        eventsUrl={mock.eventsUrl}
        scanner={mock.scanner}
      />
    );
    expect(screen.findByLabelText(mock.latestEscrow)).toBeTruthy();
  });
});

describe('when rendered with a `eventsUrl` prop', () => {
  it('should paste it into the component', () => {
    render(
      <EscrowFactoryView
        count={mock.count}
        address={mock.address}
        latestEscrow={mock.latestEscrow}
        eventsUrl={mock.eventsUrl}
        scanner={mock.scanner}
      />
    );
    expect(screen.findByLabelText(mock.eventsUrl)).toBeTruthy();
  });
});

describe('when rendered with a `scanner` prop', () => {
  it('should paste it into the component', () => {
    render(
      <EscrowFactoryView
        count={mock.count}
        address={mock.address}
        latestEscrow={mock.latestEscrow}
        eventsUrl={mock.eventsUrl}
        scanner={mock.scanner}
      />
    );
    expect(screen.findByLabelText(mock.scanner)).toBeTruthy();
  });
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <EscrowFactoryView
        count={mock.count}
        address={mock.address}
        latestEscrow={mock.latestEscrow}
        eventsUrl={mock.eventsUrl}
        scanner={mock.scanner}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
