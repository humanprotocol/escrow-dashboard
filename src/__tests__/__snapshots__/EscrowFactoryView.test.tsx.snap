// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-n4pdsy-MuiDivider-root"
      role="separator"
    >
      <span
        className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Address
        </p>
      </span>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 css-1ypfif-MuiTypography-root"
    >
      63fea823365f1c81fad234abdf5a1f43
    </p>
    <div
      className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-n4pdsy-MuiDivider-root"
      role="separator"
    >
      <span
        className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Latest Escrow
        </p>
      </span>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 css-1ypfif-MuiTypography-root"
    >
      eaac4d45c3c41f449cf7c94622afacbc
    </p>
    <div
      className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-n4pdsy-MuiDivider-root"
      role="separator"
    >
      <span
        className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Amount of jobs
        </p>
      </span>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 css-1ypfif-MuiTypography-root"
    >
      10
    </p>
    <div
      className="MuiBox-root css-xi606m"
    >
      <div
        className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-n4pdsy-MuiDivider-root"
        role="separator"
      >
        <span
          className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            All deployed escrows
          </p>
        </span>
      </div>
      <a
        className="MuiTypography-root MuiTypography-inherit MuiTypography-alignCenter MuiLink-root MuiLink-underlineAlways css-8wztv5-MuiTypography-root-MuiLink-root"
        href="test.url"
        onBlur={[Function]}
        onFocus={[Function]}
        rel="noreferrer"
        target="_blank"
      >
        test-scanner
      </a>
      <p
        className="MuiTypography-root MuiTypography-body2 css-1ncpc4g-MuiTypography-root"
      >
        Each event has a payload of ERC20 token address and Escrow Address
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 css-1ncpc4g-MuiTypography-root"
      >
        Change the type of the second argument to "Address" to see an Escrow address
      </p>
    </div>
  </div>
</div>
`;
